common/src/main/java/io/druid/math/expr/Expr.java

```diff
@@ -124,7 +123,7 @@ public ExprEval eval(ObjectBinding bindings)
 
   public StringExpr(String value)
   {
-    this.value = Strings.emptyToNull(value);
+    this.value = value;
```

Comment:
I think this will lead to incorrect behavior for the "use default values for null" case. We want nulls and empty strings to be treated equivalently: i.e. `null == ''` should return `true`, but I believe that will not be the case with the current code in this patch.

There are two general ways I could think to achieve it.

1. Ensure that `null` is never returned as the `value()` or `asString()` of an ExprEval, or as the `getLiteralValue()` of an Expr, when null-handling behavior is set to "use default values". Instead, these should all be empty strings. It could be achieved by making functions like `ExprEval.of`, `ExprEval.bestEffortOf`, `eval` methods of Expr subclasses, etc, return a StringExpr of `""` (empty string) rather than null if we are in the "use default values" null-handling mode.
2. Alternatively, allow both `null` and `""` as values, but modify the function implementations to treat them differently based on the null-handling mode.

I think (2) will work better, since in the "use default values" null-handling mode, consider that we want empty strings to be "null-like" sometimes and "empty-like" sometimes. A specific example is that in "use default values" mode, `nvl('', 'x')` should return `'x'` (the empty string is null-like). But `concat('x','','y')` should return `'xy'` (it's empty-like).

Please also add more tests around `null` and `''` in expressions as part of `EvalTest` and `ExprMacroTest`. I think that currently some behaviors are broken.